UI/js/consumeRequests.js

```diff
@@ -0,0 +1,183 @@
+
+
+let requests = {};
+const user = window.location.href.includes('engineer') ? 'engineer' : 'client';
+let requestList = {};
+let statusDiv;
+function addRequestToList(request) {
+  const { id } = request;
+
+  const updateLink = `update.html?id=${id}`;
+
+  const li = document.createElement('li');
+  if (user === 'client') {
+    let firstAnchor = `<a  class = "pending" href="${updateLink}" >Update</a>`;
+    if (request.status !== 'pending') firstAnchor = `<a  class = "${request.status}" >${capitalise(request.status)}</a>`;
+    li.innerHTML = `
+                    ${request.title}
+                    <span>
+                      
+                        <a onclick = "showRequest(event, ${id}, 'user')" href="">Details</a>        
+                        ${firstAnchor}
+                    </span>
+                `;
+  } else {
+    li.innerHTML =
+          `
+              ${request.title}
+              <span>
+              <a onclick = "showRequest(event, ${id})" href="">Details</a>        
+                  <a  class = "${request.status}" >${capitalise(request.status)}</a>
+                 
+              </span>
+          `;
+  }
+
+  requestList.appendChild(li);
+}
+
+
+function handleFetchSuccess(response, regex) {
+  if (response.statusCode === 200) {
+    requests = response.data;
+    let arr = requests;
+    if (regex) {
+      arr = requests.filter(request => regex.test(request.title));
+    }
+
+    if (arr.length === 0) {
+      const li = document.createElement('li');
```

Comment:
Instead of creating this element over and over again. I'm thinking you can just have it in the dom and just change the innerHTML everytime you want to update it.

Reply:
I don't fully understand but are you saying that I should clone a DOM instead of always creating a new one.